libs/blocks/footer/footer.js

```diff
@@ -197,23 +197,38 @@ class Footer {
 
   decoratePrivacy = () => {
     const copyrightEl = this.body.querySelector('div > p > em');
-    const links = copyrightEl?.parentElement.querySelectorAll('a');
-    if (!copyrightEl || !links) return null;
+    if (!copyrightEl) return null;
```

Comment:
Could get this out of the way so you don't have to check in your conditional
```suggestion
    const container = copyrightEl.closest('div');
    const links = container.querySelectorAll('a');
    let ul = container.querySelector('ul');
    if (!copyrightEl || !(ul || links)) return null;
```

Reply:
Not sure I see what you're saying. `container` will throw an error is `copyrightEl` is null

Reply:
You return null in the `ul` conditional, lines 222-223, if there are no `links`, so checking at the top if there are ether `links` or a `ul` and return early if both are missing.
Optional chaining for `copyrightEl` and `container`
```
    const container = copyrightEl?.closest('div');
    const links = container?.querySelectorAll('a');
    let ul = container?.querySelector('ul');
```

Reply:
Chris M has pointed out before that he prefers a conditional rather than optional chaining if you need to check the same variable multiple times. I was trying to follow his guidelines

Reply:
Returning early is also a good practice, and here I don't think we need to sacrifice one for another. It feels like optional chaining would be okay here in order to simplify the return condition, instead of how it's often over used to hunt for nested object properties but maybe that's just preference. 

If you wanted to avoid all nested chaining you could try this:
```suggestion
    if (!copyrightEl) return null;
    const container = copyrightEl.closest('div');
    if (!container) return null;
    const links = container.querySelectorAll('a');
    let ul = container.querySelector('ul');
    if !(ul || links) return null;
```
If you saying that using `container?.` twice is the issues you could try this:
```suggestion
    const container = copyrightEl?.closest('div');
    if (!copyrightEl || !container) return null;
    const links = container.querySelectorAll('a');
    let ul = container.querySelector('ul');
    if !(ul || links) return null;
```